wallets/src/main/java/network/misq/wallets/bitcoind/BitcoindProcess.java

```diff
@@ -0,0 +1,98 @@
+package network.misq.wallets.bitcoind;
+
+import network.misq.wallets.NetworkType;
+import network.misq.wallets.rpc.config.RpcConfig;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+
+public class BitcoindProcess {
+
+    private final RpcConfig rpcConfig;
+    private final Path dataDir;
+
+    private Process bitcoindProcess;
+
+    public BitcoindProcess(RpcConfig rpcConfig, Path dataDir) {
+        this.rpcConfig = rpcConfig;
+        this.dataDir = dataDir;
+    }
+
+    public void startAndWaitUntilReady() {
+        dataDir.toFile().mkdirs();
+        bitcoindProcess = createAndStartProcess();
+        waitUntilReady();
+    }
+
+    public void stopAndWaitUntilStopped() {
+        int exitCode = 1;
+        try {
+            while (exitCode != 0) {
+                Process p = new ProcessBuilder(
+                        "bitcoin-cli",
+                        rpcConfig.networkType() == NetworkType.REGTEST ? "-regtest" : "",
+                        "-rpcuser=" + rpcConfig.user(),
+                        "-rpcpassword=" + rpcConfig.password(),
+                        "stop"
+                ).start();
+                exitCode = p.waitFor();
```

Comment:
Is the while loop with the exitCode needed? The `waitFor` is blocking until the process is completed, now?